src/main/java/bio/terra/service/profile/google/GoogleBillingService.java

```diff
@@ -71,39 +75,41 @@ private static Cloudbilling cloudbilling() {
      * @return true if the repository can act as a billing account *user* (viewer is not enough), false otherwise
      */
     public boolean canAccess(BillingProfileModel billingProfile) {
-        String accountId = "billingAccounts/" + billingProfile.getBillingAccountId();
+        String accountId = billingProfile.getBillingAccountId();
+        ResourceName resource = BillingAccountName.of(accountId);
         List<String> permissions = Collections.singletonList("billing.resourceAssociations.create");
-        TestIamPermissionsRequest permissionsRequest = new TestIamPermissionsRequest().setPermissions(permissions);
+        TestIamPermissionsRequest permissionsRequest = TestIamPermissionsRequest.newBuilder()
+            .setResource(resource.toString())
+            .addAllPermissions(permissions)
+            .build();
         try {
-            Cloudbilling.BillingAccounts.TestIamPermissions testIamPermissions =
-                cloudbilling().billingAccounts().testIamPermissions(accountId, permissionsRequest);
-            List<String> actualPermissions = testIamPermissions.execute().getPermissions();
+            TestIamPermissionsResponse response = cloudBillingClient().testIamPermissions(permissionsRequest);
+            List<String> actualPermissions = response.getPermissionsList();
             return actualPermissions != null && actualPermissions.equals(permissions);
-        } catch (GoogleJsonResponseException e) {
-            int status = e.getStatusCode();
+        } catch (ApiException e) {
+            int status = e.getStatusCode().getCode().getHttpStatusCode();
             if (status == 400 || status == 404) {
                 // The billing account id is invalid or does not exist. This counts as inaccessible.
                 return false;
             }
-            String message = String.format("%s Error, Could not check permissions on: %s", status, accountId);
-            throw new BillingServiceException(message, e);
-        } catch (IOException e) {
-            String message = String.format("Could not check permissions on: %s", accountId);
+            String message = String.format("ERROR: Could not check permissions on `%s`: %s",
+                resource.toString(),
+                e.getMessage());
```

Comment:
If you do this then `e.Message()` will be printed twice in the stack crawl; once at this level and another time at the level that it occurs (further down/up the stack). It's OK to print it twice but it seems redundant to me.

Reply:
Good point! I fixed this. Thank you!